### ISSUE TITLE
t.match is not a function in prod deployement

### ISSUE THREAD

**Issue by AMontagu** [angular/angular#12007]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => search github for a similar issue or PR before submitting
```

**Current behavior** 

I am currently working on a project based on [https://github.com/AngularClass/angular2-webpack-starter](https://github.com/AngularClass/angular2-webpack-starter). Everything is working fin in dev deployement (npm run server:dev ).

But after a `npm run build:prod` and `npm run server:prod` when I try to access to my app I have this js error : 

[http://pasteboard.co/9llDiJNGv.png](http://pasteboard.co/9llDiJNGv.png)

```
function _split(t) {
    return t.match(c)
}
```

Currently after pretty print in chrome the error seem to be an error in my component routing somewhere. I will investigate more but I post this issue because is not normal that my code is working in dev but not in prod. If I find why I will post the answers here.

Currently my routing look like: 
 app.routes.ts:
```
export const ROUTES: Routes = [
  { path: '', loadChildren: './app/<componentName>/<componentName>.module#<componentName>Module' },
  { path: '**',    component: NoContent },
];
```

\<componentName\>.routing.ts:
```
const routes: Routes = [
  { path: "",
    component: <componentName>Component,
    children: [
      { path:"", component: <componentName>ListComponent},
      { path:"<componentName2>", component: <componentName2>},
      { path:"<componentName3>", component: <componentName3>},
        { path:"<componentName4>", component: <componentName4>},
        { path:"<componentName5>", component: <componentName5>},
    ]
  }
];
```
 

**Expected behavior**
No javascript error and my app working like in dev mode

**Reproduction of the problem**
Can't see how.

**What is the motivation / use case for changing the behavior?**
Having a working app

**Please tell us about your environment:**
All operating system
IDE -> webstorm 2016.2.3
package manager -> npm
HTTP server -> node (windows) and nginx(linux)

* **Angular version:** 2.0.0

* **Browser:** [all ] 
 
* **Language:** [TypeScript 2.0.3]
 
* **Node (for AoT issues):** `node --version` =   v6.6.0 and v6.7.0

**Comment by ericmartinezr** [comment]
You're seeing this error

https://github.com/angular/angular/issues/10626
https://github.com/angular/angular/issues/11590

**Comment by vicb** [comment]
thanks @ericmartinezr should be closed by #11617

**Comment by AMontagu** [comment]
Thank you ! Indeed removing module.id was the tricks.

Sorry for the duplicate. When I searched I thought is was a different error.
